lib/xandra/connection.ex

```diff
@@ -651,30 +648,30 @@ defmodule Xandra.Connection do
     {:keep_state, data}
   end
 
-  def connected(:cast, {:timed_out_id, stream_id}, %__MODULE__{} = data) do
+  # The caller is notifying the connection that the request (on stream_id) timed
+  # out on its side (that is, the caller reached its "after" in the receive block).
+  # We need to remove the stream ID from the in-flight requests but still keep
+  # track of it, because C* might still send us a response for that query (and when it
+  # does, we will throw it away and free the timed-out stream ID).
+  #
+  # C* does not support CANCELING queries, by the way, otherwise that's what we'd do here.
+  def connected(:cast, {:request_timed_out_at_caller, stream_id}, %__MODULE__{} = data) do
     data = update_in(data.in_flight_requests, &Map.delete(&1, stream_id))
-    data = update_in(data.timed_out_ids, &Map.put(&1, stream_id, DateTime.utc_now()))
-
+    data = put_in(data.timed_out_ids[stream_id], System.system_time(:millisecond))
     {:keep_state, data}
   end
 
-  def connected({:timeout, :flush_timed_out_stream_id}, :flush, %__MODULE__{} = data) do
-    now = DateTime.utc_now()
+  def connected({:timeout, :flush_timed_out_stream_ids}, _content, %__MODULE__{} = data) do
+    now = System.system_time(:millisecond)
```

Comment:
Timestamp is cheaper than a whole `DateTime` struct, which we really don't need.